sw/host/tests/rom/e2e_bootstrap_rma/src/main.rs

```diff
@@ -18,362 +21,322 @@ use opentitanlib::dif::lc_ctrl::{
+            .context("failed to build regex")?;
+    let exit_failure =
+        Regex::new("reset_info_bitfield: 0x[0-9a-f]+\r\n").context("failed to build regex")?;
+
+    let mut console = UartConsole {
+        timeout: Some(CONSOLE_TIMEOUT),
+        exit_success: Some(exit_success),
+        exit_failure: Some(exit_failure),
+        ..Default::default()
     };
-    use OpenocdTclBlock::{AssertRegEq, Echo, PollUntilRegEq, WriteReg};
 
+    let mut stdout = io::stdout();
+    let result = console
+        .interact(&*uart, None, Some(&mut stdout))
+        .context("failed to interact with console")?;
+
+    match result {
+        ExitStatus::ExitSuccess => Ok(()),
+        result => bail!("FAIL: {result:?}"),
+    }
+}
+
+/// Iteration 2 of the `rom_bootstrap_rma` testpoint.
+///
+/// Issues the lifecycle RMA command and ensures the RMA transition is completed
+/// successfully.
+fn test_rma_command(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    let rma_bootstrap_strapping = transport.pin_strapping("RMA_BOOTSTRAP")?;
+    let tap_strapping = transport.pin_strapping("PINMUX_TAP_LC")?;
+
+    // Need to reset with `RMA_BOOTSTRAP` set.
+    rma_bootstrap_strapping
+        .apply()
+        .context("failed to apply RMA_BOOTSTRAP strapping")?;
+
+    // We keep `PINMUX_TAP_LC` set so that resets due to transition don't
+    // disconnect us from the LC TAP.
+    tap_strapping
+        .apply()
+        .context("failed to apply PINMUX_TAP_LC strapping")?;
+
+    transport
+        .reset_target(opts.init.bootstrap.options.reset_delay, true)
+        .context("failed to reset target")?;
+
+    // Remove the RMA_BOOTSTRAP strapping to bring the device out of the RMA spin loop.
+    rma_bootstrap_strapping
+        .remove()
+        .context("failed to remove strapping RMA_BOOTSTRAP")?;
+
+    // NOTE: without this sleep, the transition starts but isn't successful.
+    // The sleep must happen anywhere before the `START` `TRANSITION_CMD` is given.
+    // Not sure why at the moment, seems like the chip isn't ready even though the
+    // `STATUS.READY` bit is checked.
+    std::thread::sleep(Duration::from_secs(1));
+
+    log::info!("Connecting to JTAG interface");
+
+    let jtag = transport.jtag(&opts.jtag).context("failed to get JTAG")?;
+    jtag.connect(JtagTap::LcTap)
+        .context("failed to connect to JTAG")?;
+
+    // Wait for the lifecycle controller to enter the `READY` state from
+    // which it accepts transition commands.
+    wait_until_ready(&jtag, Duration::from_secs(3))
+        .context("failed to wait for lifecycle controller to be ready")?;
+
+    // Check we're in the `PROD` state with 5 transitions registered.
+    assert_eq!(
+        jtag.read_lc_ctrl_reg(&LcCtrlReg::LcState)?,
+        DifLcCtrlState::Prod.redundant_encoding()
+    );
+    assert_eq!(jtag.read_lc_ctrl_reg(&LcCtrlReg::LcTransitionCnt)?, 5);
```

Comment:
You don't need this assertion, you just want to know that there is an available amount of transition left the device can perform. Using the `trigger_lc_transition` testutil I mentioned above should eliminate this anyway, and eliminate the stale OTP image on the FPGA between consecutive runs you described.

Reply:
I'll remove this assertion if you want, but it did help me debug (in further runs) the issue with the failing transition which was incrementing `TRANSITION_CNT` but not transitioning from `PROD` nor setting any error bits.

Reply:
Ah I see, thanks for clarifying. It's ok to keep it then.